### DISCUSSION TITLE
Did anyone manage to set a Regex for files in a specific path?

### DISCUSSION THREAD

**Discussion by ololiuhqui** [FlorianWoelki/obsidian-icon-folder#199] [Q&A]
I tried something like `.*\/folder($|\/.*)` (with and without escaping the backslash).

But while in https://regex101.com/ is showing to be matching all files and folders within that specific path, it does not in Obsidian, am i missing something?

It looks like only simple global rules like `.*` do work for me.

**Comment by ReWaXx** [comment]
After much testing, I managed to match a few custom rules: 
.* -> match all files and folder
.\*.md -> match all markdown files inside the vault (also works with png, pdf)
folder/* -> match all files inside "folder" (folder is at the root of the vault)

**Reply by loganmancuso** [reply]
is there any way to match any folder like **/ or equivalent?

**Reply by CHerSun** [reply]
- list your vault file tree
- copy & paste that into regex101.com
- try patterns you want.

I'd highly recommend any regex cheatsheet

`.*` - just matches absolutely any string
`.*.md` - matches any string of at least 3 characters, which contains `md` text in it, not necessarily the extension. For extension I'd use something like `.*\.md$`
`folder/*` is normally an incorrect pattern, symbol `/` needs escaping to work properly, but if specific regex engine can use it directly, then the pattern would match any string, which contains symbols `folder` anywhere in it (`*` being a quantifier of 0...any number of previous entity).
